connector/core/core-service/src/main/java/org/oaebudt/edc/core/store/SqlParticipantGroupStore.java

```diff
@@ -0,0 +1,129 @@
+                                    TransactionContext transactionContext,
+                                    ObjectMapper objectMapper,
+                                    QueryExecutor queryExecutor) {
+        super(dataSourceRegistry, dataSourceName, transactionContext, objectMapper, queryExecutor);
+    }
+
+    @Override
+    public ParticipantGroup findById(String groupId) {
+        return transactionContext.execute(() -> {
+            try (var connection = getConnection()) {
+                Set<String> participantIds = queryExecutor.query(connection, false, this::mapParticipant, this.findParticipantsByGroupId(), groupId)
+                        .collect(Collectors.toSet());
+
+                return new ParticipantGroup(groupId, participantIds);
+            } catch (SQLException e) {
+                throw new EdcPersistenceException(e);
+            }
+        });
+    }
+
+    @Override
+    public void save(String groupId, String... participantIds) {
+        transactionContext.execute(() -> {
+            try (var connection = getConnection()) {
+                queryExecutor.execute(connection, getUpsertParticipantGroupStatement(), groupId);
+
+                if(participantIds.length > 0) {
+                    queryExecutor.execute(connection, getUpsertMultipleParticipantStatement(groupId, participantIds));
+                }
+            } catch (SQLException e) {
+                throw new EdcPersistenceException(e);
+            }
+        });
+    }
+
+    @Override
+    public Set<ParticipantGroup> getAllGroups() {
+
+        return transactionContext.execute(() -> {
+
+            try (var connection = getConnection()) {
+
+                Map<String, Set<String>> grouped =queryExecutor.query(connection, false, this::mapParticipantGroup, this.findAllParticipants())
+                        .collect(Collectors.groupingBy(
+                                arr -> arr[0], // key
+                                Collectors.mapping(
+                                        arr -> arr[1], // value
+                                        Collectors.filtering(
+                                                Objects::nonNull,
+                                                Collectors.toSet()
+                                        )
+                                )
+                        ));
+
+                Set<ParticipantGroup> result = new HashSet<>();
+                for (Map.Entry<String, Set<String>> entry : grouped.entrySet()) {
+                    result.add(new ParticipantGroup(entry.getKey(), entry.getValue()));
+                }
+                return result;
+            } catch (SQLException e) {
+                throw new EdcPersistenceException(e);
+            }
+        });
+    }
+
+
+    public String getUpsertParticipantGroupStatement() {
+        return "INSERT INTO %s (%s) VALUES (?) ON CONFLICT DO NOTHING;".formatted(PARTICIPANT_GROUP_TABLE, "id");
```

Comment:
I don't think the "on conflict do nothing" is equivalent to an upsert operation ([ref.](https://www.postgresql.org/docs/current/sql-insert.html))
please verify the upsert behavior with a test